utils/fixed_decimal/src/decimal.rs

```diff
@@ -1300,32 +1300,220 @@ impl FixedDecimal {
+                        *digit = 5;
+
+                        // Early return since the expansion didn't generate a remainder.
+                        #[cfg(debug_assertions)]
+                        self.check_invariants();
+                        return;
+                    }
+
+                    *digit = 10;
+
+                    iter
+                }
+                RoundingIncrement::MultiplesOf25 => {
+                    // Extend the digits to have the correct
+                    // number of trailing zeroes.
+                    self.digits.resize(digits_to_retain as usize, 0);
+
+                    let Some((last_digit, digits)) = self.digits.split_last_mut() else {
+                        debug_assert!(false, "`self.digits` should have at least a digit");
+                        return;
+                    };
+
+                    let Some((second_last_digit, _)) = digits.split_last_mut() else {
+                        // The number has no other digits aside from the last.
+                        // We need to manually increment the number to 25
+
+                        if self.magnitude == i16::MAX {
+                            overflow(self);
+                            return;
+                        }
+
+                        self.digits.clear();
+                        self.digits.extend_from_slice(&[2, 5]);
+                        self.magnitude += 1;
+                        self.upper_magnitude = cmp::max(self.upper_magnitude, self.magnitude);
+
+                        #[cfg(debug_assertions)]
+                        self.check_invariants();
+                        return;
+                    };
+
+                    let number = *second_last_digit * 10 + *last_digit;
+
+                    if number < 75 {
+                        // We can return directly if the number won't expand
+                        // to 100.
+                        if number < 25 {
+                            *second_last_digit = 2;
+                            *last_digit = 5;
+                        } else if number < 50 {
+                            *second_last_digit = 5;
+                            self.digits.pop();
+                        } else {
+                            *second_last_digit = 7;
+                            *last_digit = 5;
+                        }
+
+                        #[cfg(debug_assertions)]
+                        self.check_invariants();
+                        return;
+                    }
+
+                    // The number reached 100. Continue the algorithm but with the
+                    // last two digits removed from the iterator.
+
+                    *second_last_digit = 10;
+                    *last_digit = 10;
+
+                    let mut iter = self.digits.iter_mut().rev().enumerate();
+
+                    // Remove the last two items that were already handled.
+                    iter.next();
+                    iter.next();
```

Comment:
Suggestion (optional): to be more consistent with the other branches, consider using iterator methods to fetch `last_digit` and `second_last_digit` instead of `split_last_mut`. This is optional because what you currently have written is more consistent with how the `trunc` function is written.

Reply:
I think I thought about doing it like that, but I opted for this because `iter` doesn't allow having two mutable references to its elements, so we'd have to recreate the iterator multiple times to first get the full number, then to modify the existing digits.